Test failing on Windows

```
 C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\zarr\storage.py:1144: KeyError
____________________________ test_write_read_tiles ____________________________

self = <zarr.storage.DirectoryStore object at 0x00000207BC15B510>
key = 'LPT1/.zarray'
value = array([123,  10,  32,  32,  32,  32,  34,  99, 104, 117, 110, 107, 115,
        34,  58,  32,  91,  10,  32,  32,  32,...32,  32,  34, 122,  97, 114, 114,  95, 102, 111, 114,
       109,  97, 116,  34,  58,  32,  50,  10, 125], dtype=uint8)

    def __setitem__(self, key, value):
        key = self._normalize_key(key)
    
        # coerce to flat, contiguous array (ideally without copying)
        value = ensure_contiguous_ndarray_like(value)
    
        # destination path for key
        file_path = os.path.join(self.path, key)
    
        # ensure there is no directory in the way
        if os.path.isdir(file_path):
            shutil.rmtree(file_path)
    
        # ensure containing directory exists
        dir_path, file_name = os.path.split(file_path)
        if os.path.isfile(dir_path):
            raise KeyError(key)
        if not os.path.exists(dir_path):
            try:
>               os.makedirs(dir_path)
E               NotADirectoryError: [WinError 267] The directory name is invalid: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpyd66wm0f\\tiled.zarr\\LPT1'

C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\zarr\storage.py:1141: NotADirectoryError
```

Possibly related to #255?

--- a/tests/ngff/test_ngff.py
+++ b/tests/ngff/test_ngff.py
@@ -50,6 +50,16 @@ short_alpha_numeric = st.text(
     min_size=1,
     max_size=16,
 )
+if os.name == "nt":
+    # Windows does not allow certain file names
+    _INVALID_NT_FILE_NAMES = (
+        ["CON", "PRN", "AUX", "NUL"]
+        + ["COM" + str(i) for i in range(10)]
+        + ["LPT" + str(i) for i in range(10)]
+    )
+    short_alpha_numeric = short_alpha_numeric.filter(
+        lambda x: x not in _INVALID_NT_FILE_NAMES
+    )
 tiles_rc_st = st.tuples(t_dim_st, t_dim_st)
 plate_axis_names_st = st.lists(
     short_alpha_numeric,

